docs/plugin/messaging.md

```diff
@@ -0,0 +1,886 @@
+In the past, prior to the R3 version of Thunder, logging, tracing and warning reporting were completely separate issues and were handled individually. However, even though these message types have distinct characteristics, we recognized the advantages of consolidating them into a unified framework, which is now referred to as `Messaging`. In the early versions of R3, we still had tracing enabled by default and messaging was in the early development. It is the R4 version of Thunder where `Messaging` is finally the default framework to handle all types of messages.
+
+
+# Advantages of using messaging
```

Comment:
There's a quirk in the mkdocs theme we're using where the table of contents doesn't work if you use H1 (`#`) headings in the documentation.

All headings should be H2 (`##`) or smaller, so can you update to change this? If it's right, you'll see a Table of Contents appear on the right hand side of the page :)

Reply:
So that's why the table of contents was not showing, it makes sense now. Thanks, I'll update it and remember that for the future docs 😃